src/rpc/misc.cpp

```diff
@@ -372,6 +383,15 @@ UniValue verifymessage(const JSONRPCRequest& request)
     if (!pubkey.RecoverCompact(ss.GetHash(), vchSig))
         return false;
 
+    //Possibly p2sh-p2wpkh
+    if (scriptID != CScriptID()) {
+        CScriptID nestedScriptID = CScriptID(CScript() << OP_0 << ToByteVector(pubkey.GetID()));
+        //Failure here could mean address is any other type of p2sh
+        if (scriptID != nestedScriptID) {
+            throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Given P2SH address does not match the signature.");
+        }
+    }
+
```

Comment:
I'm not sure we want to extend the current message signing to new address formats as-is. Maybe better to leave this out for now...